lib/matplotlib/legend.py

```diff
@@ -1269,6 +1278,44 @@ def _find_best_position(self, width, height, renderer, consider=None):
     def contains(self, event):
         return self.legendPatch.contains(event)
 
+    def set_draggable(self, state, use_blit=False, update='loc'):
+        """
+        Enable or disable mouse dragging support of the legend.
+
+        Parameters
+        ----------
+        state : bool
+            ``True`` / ``False`` enables / disables mouse dragging.
+        use_blit : bool, optional
+            Blitting is an old and basic but fast composition method for
+            images. Usually, you won't need to set this. However, you may use
+            this at the cost of a reduced quality if you have performance
+            issues when dragging. See also :ref:`func-animation`.
+        update_loc : ['loc' | 'bbox'], optional
+            The legend parameter to be changed when dragged:
+
+            - 'loc': update the *loc* parameter of the legend
+            - 'bbox': update the *bbox_to_anchor* parameter of the legend
+
+        Returns
+        -------
+        If *state* is ``True`` this returns the `~.DraggableLegend` helper
+        instance. Otherwise this returns ``None``.
+        """
+        if state:
+            if self._draggable is None:
+                self._draggable = DraggableLegend(self,
+                                                  use_blit,
+                                                  update=update)
+        else:
+            if self._draggable is not None:
+                self._draggable.disconnect()
+            self._draggable = None
+
+    def get_draggable(self):
+        """Return ``True`` if the legend is draggable, ``False`` otherwise."""
+        return self._draggable is not None
```

Comment:
I'm not quite following what happened here with `self._draggable`: `draggable()` used to return `self._draggable`.  `.get_draggable` returns Bool.  I'm not sure what the point of returning the actual object was (isn't it `self`??), but now we are returning something new.  Can we still get at `self._draggable` if someone needs to?

Reply:
The API of `draggable()` is unchanged. There was a missing return statement in `set_draggable()`. With arguments it behaves now exactly the same as `draggable()` with arguments (returns `DraggableLegend` or `None`). There hasn't been an equivalent for `get_draggable` so far. Neither was there a way to check if dragging is enabled, nor could you get the `DraggableLegend` instance, other than accessing `_draggable`. Note: `DraggableLegend` is a wrapper around a `Legend`.